src/components/MintDialog/elements/NativeMintButton.tsx

```diff
@@ -53,43 +55,51 @@ export const NativeMintButton: FC<NativeMintButtonProps> = ({
   const mint = useMemo(() => {
     switch (mintType) {
       case MintType.Zora:
-        return () => {
+        return async () => {
           if (!account) {
             setPage(ModalPage.MINT_ERROR)
+            return
           }
-          return writeZora721({
+          return (await writeZora721({
             address: address,
             functionName: 'mintWithRewards',
-            args: [account!, BigInt(quantity), '', getAddress(creatorAddress)],
+            args: [account, BigInt(quantity), '', getAddress(creatorAddress)],
             value: price,
-          })
+          })).hash
         }
       case MintType.ThirdWeb:
-        return () =>
-          writeTw721({
-            address: address as Address,
-            functionName: 'claim',
-            chainId: l2.id,
-            args: [
-              account || '0x0',
-              BigInt(quantity),
-              '0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE',
-              price / BigInt(quantity),
-              {
-                proof: [],
-                quantityLimitPerWallet: 2n ** 256n - 1n,
-                pricePerToken: price / BigInt(quantity),
-                currency: '0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE',
-              },
-              '0x0',
-            ],
-            value: price,
-          })
+        return async () => {
+          if (!account || !walletClient || !publicClient) {
+            setPage(ModalPage.MINT_ERROR)
+            return
+          }
+          const {request} = await publicClient.simulateContract({
+            abi: tw721ABI,
+              address: address as Address,
+              functionName: 'claim',
+              args: [
+                account,
+                BigInt(quantity),
+                '0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE',
+                price / BigInt(quantity),
+                {
+                  proof: [],
+                  quantityLimitPerWallet: 2n ** 256n - 1n,
```

Comment:
This seems worth naming too

Reply:
yeah agree, this is all a bit jank here. Was just moving this into a new function call as the wagmi cli generated stuff was dropping the datasuffix